rfd/0219-k8s-rbac-simplification.md

```diff
@@ -0,0 +1,280 @@
+Originally, Teleport preset roles were inspired by Kubernetes roles,
+specifically, `edit`, `cluster-admin` and `view`.
+
+- `cluster-admin` allows everything everywhere.
+- `edit` allows CRUD on most builtin resources, but not administrative
+  resources.
+  - get/list/watch pods (including attach, exec, portforward and proxy),
+    secrets and services/proxy
+  - impersonate service accounts
+  - create/delete/deletecollection/patch/update pods (including exec/attach/
+    portforward/proxy)
+  - create pods/evictions
+  - create/delete/deletecollection/patch/update configmaps, events,
+    persistentvolumeclaims, replicationcontrollers (including scale), secrets,
+    serviceaccounts, services (including proxy)
+  - create serviceaccounts/token
+  - create/delete/deletecollection/patch/update deployments (including
+    rollback, scale), replicasets (including scale), statefulsets (including
+    scale), ingresses, networkpolicies
+  - create/delete/deletecollection/patch/update horizontalpodautoscalers, cronjobs,
+    jobs, poddisruptionbudgets, leases
+  - get/list/watch controllerrevisions, daemonsets, deployments, replicasets,
+    statefulsets, horizontalpodautoscalers, cronjobs, jobs, poddisruptionbudgets,
+    ingresses, networkpolicies, leases
+- `view` allows read-only access to most builtin resources.
+  - get/list/watch pods, configmaps, endpoints, persistentvolumeclaims,
+    replicationcontrollers (including scale), serviceaccounts, services,
+    bindings, events, limitranges, namespaces, resourcequotas, endpointslices,
+    controllerrevisions, daemonset, deployments, replicasets, statefulsets,
+    horizontalpodautoscalers, cronjobs, jobs, poddisruptionbudgets, ingresses,
+    networkpolicies, leases
+
+This resulted in the following Teleport roles:
+
+- `editor`, which maps to `cluster-admin` and can do everything.
+- `access`, which maps to `edit` and can do most CRUD operations.
+- `auditor`, which maps to `view` and can do read-only operations.
+
+#### New Kubernetes Bindings
+
+In order to be consistent and allow for seamless use of Kubernetes within
+Teleport, we will introduce the following new preset bindings:
+
+- `kubernetes_groups` -> `ClusterRole`
+- `teleport:preset:edit` -> `edit`
+  Can CRUD most basic builtin resources and read some limited administrative resources.
+- `teleport:preset:cluster-admin` -> `cluster-admin`
+  Provides full access to all Kubernetes resources, including CRDs.
+- `teleport:preset:view` -> `view`
+  Provides read-only access to some Kubernetes resources.
+
+On the Kubernetes side, we will create the corresponding `ClusterRoleBindings` based on `kubernetes_groups`.
+
+While there is an existing binding for `cluster-admin` via `system:masters`, it
+is not available everywhere and there are no binding for `edit` nor `view`.
+For consistency and reliability, we will create one binding for each preset
+role.
+
+Those bindings will be available to Teleport users via the user trait feature.
+
```

Comment:
Adding more details from the discussions with @klizhentas about why we prefer to not do this ourselves.

```suggestion

Kubernetes might introduce new API endpoints that should be allowed or denied.
Vendoring those roles on Teleport would require us to update the list for every
Kubernetes version, and would potentially cause a permission issues if a user
deploys an old Teleport version in a modern Kubernetes cluster. Relying on the
Kubernetes built-in roles for RBAC frees us from the maintenance burden and
prevents security issues caused by version mismatch.

```